spec/models/match_spec.rb

```diff
@@ -35,4 +35,32 @@
     expect(match_b.away_matches).to be_empty
     expect(match_b.matches).to contain_exactly(match_a)
   end
+  
+  describe '#calculate_points' do
+    subject { -> { match.calculate_points } }
+
+    context 'no team has a score' do
+      let(:match) { FactoryBot.build(:match, :empty_points, score_home: nil, score_away: nil) }
+      it { is_expected.not_to change { match.points_home } }
+      it { is_expected.not_to change { match.points_away } }
+    end
+
+    context 'team home has a higher score' do
+      let(:match) { FactoryBot.build(:match, :empty_points, score_home: 1, score_away: 0) }
+      it { is_expected.to change { match.points_home }.from(nil).to(3) }
+      it { is_expected.to change { match.points_away }.from(nil).to(0)  }
+    end
+
+    context 'team away has a higher score' do
+      let(:match) { FactoryBot.build(:match, :empty_points, score_home: 0, score_away: 1) }
+      it { is_expected.to change { match.points_home }.from(nil).to(0) }
+      it { is_expected.to change { match.points_away }.from(nil).to(3)  }
+    end
+
+    context 'team home and team away have an equal score' do
+      let(:match) { FactoryBot.build(:match, :empty_points, score_home: 1, score_away: 1) }
+      it { is_expected.to change { match.points_home }.from(nil).to(1) }
+      it { is_expected.to change { match.points_away }.from(nil).to(1) }
+    end
```

Comment:
case for "manually entered points not modified" could be added

Reply:
That is not part of the model